cpu/stm32_common/periph/i2c_1.c

```diff
@@ -428,14 +443,28 @@ static inline int _write(I2C_TypeDef *i2c, const uint8_t *data, size_t length)
         uint16_t tick = TICK_TIMEOUT;
         while (!(i2c->ISR & I2C_ISR_TXIS) && tick--)  {
             if (i2c->ISR & ERROR_FLAG || !tick) {
-                return -ETIMEDOUT;
+                DEBUG("[i2c] write: TXIS timeout\n");
+                return -ENXIO;
             }
         }
-
         /* write data to data register */
-        DEBUG("[i2c] write: Write byte %i to DR\n", i);
+        DEBUG("[i2c] write: Write byte %02X to DR\n", data[i]);
         i2c->TXDR = data[i];
         DEBUG("[i2c] write: Sending data\n");
+
+        int ret = _check_bus(i2c);
+        if (ret < 0) {
+            return ret;
+        }
+    }
+
+    DEBUG("[i2c] write: Waiting for write to complete\n");
+    uint16_t tick = TICK_TIMEOUT;
+    while (!(i2c->ISR & I2C_ISR_TC) && tick--)  {
+        if (i2c->ISR & ERROR_FLAG || !tick) {
```

Comment:
the `!tick` will never happen due to the `while` condition, hence I would suggest to move the `if` after the `while`

Reply:
okay, actually it will happen, but only on the last iteration, so I would still move it after the while, looks way cleaner (IMHO)